cmd/batch-handlers.go

```diff
@@ -1174,6 +1212,25 @@ func (j *BatchJobPool) queueJob(req *BatchJobRequest) error {
 	return nil
 }
 
+// delete canceler from the map, cancel job if requested
+func (j *BatchJobPool) canceler(jobID string, cancel bool) error {
+	if j == nil {
+		return errInvalidArgument
+	}
+	j.jmu.Lock()
+	defer j.jmu.Unlock()
+	if canceler, ok := j.jobCancelers[jobID]; ok {
+		if cancel {
+			canceler()
+
+		}
+	} else {
+		return fmt.Errorf("job %s not found", jobID)
```

Comment:
We do not need to return errors for job that do not exist, treat it like delete call. 204 always..